comms/lighthouse/src/peers/initPeerJsServer.ts

```diff
@@ -0,0 +1,73 @@
+import { Express } from 'express'
+import { Server } from 'net'
+import { ExpressPeerServer } from 'peerjs-server'
+import { IConfig } from 'peerjs-server/dist/src/config'
+import { MessageType } from 'peerjs-server/dist/src/enums'
+import { LayersService } from '../layersService'
+import { PeersService } from '../peers/peersService'
+import { ArchipelagoService } from './archipelagoService'
+import { peerAuthHandler } from './auth'
+import { IdService } from './idService'
+
+export type PeerJSServerInitOptions = {
+  netServer: Server
+  idService: IdService
+  noAuth: boolean
+  peersServiceGetter: () => PeersService
+  ethNetwork: string
+  archipelagoServiceGetter: () => ArchipelagoService
+  layersServiceGetter: () => LayersService
+}
+
+export function initPeerJsServer({
+  netServer,
+  idService,
+  noAuth,
+  peersServiceGetter,
+  archipelagoServiceGetter,
+  layersServiceGetter,
+  ethNetwork
+}: PeerJSServerInitOptions): Express {
+  const options: Partial<IConfig> = {
+    path: '/',
+    idGenerator: () => idService.nextId(),
+    authHandler: peerAuthHandler({ noAuth, peersServiceGetter, ethNetwork })
+  }
+
+  const peerServer = ExpressPeerServer(netServer, options)
+
+  peerServer.on('disconnect', (client: any) => {
+    console.log('User disconnected from server socket. Removing from all rooms & layers: ' + client.id)
+    layersServiceGetter().removePeer(client.id)
+    archipelagoServiceGetter().clearPeer(client.id)
+  })
+
+  peerServer.on('error', console.log)
+
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  //@ts-ignore
```

Comment:
What is the purpose of that `//@ts-ignore`?

Reply:
LET US WORK @menduz.

This was here before and will be removed IF WE CAN

Reply:
It is because PeerJSServer types don't correspond to the actual event that you receive.

This was actually here before. We just moved it around. Eventually, we'll try to get rid of the ts-ignore, at least trying to replace them for any or something not so drastic.